botnim/collect_sources.py

```diff
@@ -1,53 +1,117 @@
 import io
-import pathlib
+from pathlib import Path
 import dataflows as DF
-
+from datetime import datetime, timezone
+import json
+from content_extraction import extract_structured_content
 
 def collect_sources_files(config_dir, context_name, source):
     files = list(config_dir.glob(source))
-    file_streams = [(f.name, f.open('rb'), 'text/markdown') for f in files]
+    file_streams = []
+    for f in files:
+        with open(f, 'rb') as ff:
+            content = ff.read()
+        file_streams.append((f.name, io.BytesIO(content), f'text/{f.suffix[1:]}', None))
     return file_streams
 
 def collect_sources_split(config_dir, context_name, source):
-    filename = config_dir / source
-    content = filename.read_text()
-    content = content.split('\n---\n')
-    file_streams = [io.BytesIO(c.strip().encode('utf-8')) for c in content]
-    file_streams = [(f'{context_name}_{i}.md', f, 'text/markdown') for i, f in enumerate(file_streams)]
+    with open(config_dir / source, 'r') as f:
```

Comment:
A general comment, not relevant only here:
why change all of the function's code when all you need is to add a single `None` to a tuple?

This is not about style - it's about code quality and maintainability.
Taking a function that already empirically works and changing it completely, makes it much harder to ensure that the change does not introduce any unwanted side effects, and thus reduces code quality.